### PR TITLE
Add input validation for AddEnv option

### LINKED ISSUE #5: Add input validation for AddEnv option
## Summary
Add input validation to the `AddEnv` function to ensure proper key/value format and prevent common errors.

## Current issue
The `AddEnv(key, value string)` function currently accepts any strings without validation, which could lead to:
- Empty keys or values
- Keys with invalid characters (=, null bytes, etc.)
- Values that could cause shell injection issues

## Proposed validation
- Validate that key is non-empty and doesn't contain '=' or null bytes
- Consider validating against common environment variable naming conventions (alphanumeric + underscore, starting with letter/underscore)
- Validate that value doesn't contain null bytes
- Return descriptive errors for invalid inputs

## Implementation considerations
- Should be backwards compatible where possible
- Consider whether to be strict (POSIX compliance) or permissive
- Add tests for various edge cases
- Update documentation with validation rules

@claude

### PR DESCRIPTION
**Pull request by broothie**
Add input validation to the `AddEnv` function to ensure proper key/value format and prevent common errors.

## Changes
- Added POSIX-compliant validation for environment variable keys
- Added validation for environment variable values
- Added comprehensive test coverage for all validation scenarios
- Updated README with validation rules and examples

## Validation Rules
**Key validation:**
- Must be non-empty
- Must start with letter or underscore
- Can only contain letters, digits, and underscores
- Cannot contain '=' or null bytes

**Value validation:**
- Cannot contain null bytes

Fixes #5

Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ cmd, err := cob.New(ctx, "echo",
 	cob.SetArgs("Hello", "World"),         // Set args directly
 	cob.AddArgs("more", "args"),           // Add to existing args
 	cob.SetEnv("SHELL=bash"),              // Set env directly
-	cob.AddEnv("SHELL", "bash"),           // Add to existing env
+	cob.AddEnv("SHELL", "bash"),           // Add to existing env (validates key/value)
```
Comment (broothie) [inline README.md:41]: @claude no need for this change

File: README.md [hunk 2]
```diff
@@ -53,6 +53,38 @@ cmd, err := cob.New(ctx, "echo",
 )
 ```
 
+## Environment Variable Validation
```
Comment (broothie) [inline README.md:56]: @claude no need for this change either
